### PR TITLE
feat: wallet balance component

### LINKED ISSUE #25: Component that displays wallet ERC20 token balances

### LINKED ISSUE #56: Component that displays wallet native balance

### PR DESCRIPTION
**Pull request by krzysu**
Closes #56 
Closes #25 

- Adds components to display wallet details and balance.
- Adds wallet page to the demo app

Screenshot 2024-12-10 at 14 30 18
Screenshot 2024-12-10 at 14 30 00

### CODE REVIEW

File: packages/web3-circle-sdk/src/FaucetApi.ts [hunk 1]
```diff
@@ -6,11 +6,11 @@ export class FaucetApi extends BaseApi {
    * Request testnet tokens for your wallet.
    * https://developers.circle.com/api-reference/w3s/programmable-wallets/request-testnet-tokens
    * @param params the parameters for the faucet request
-   * @returns boolean
+   * @returns void
+   *
+   * TODO: this endpoint doesn't work, returns TypeError: Only absolute URLs are supported
```
Comment (krzysu) [inline packages/web3-circle-sdk/src/FaucetApi.ts:11]: fyi @avkos

File: packages/web3-circle-sdk/src/WalletApi.ts [hunk 2]
```diff
@@ -62,9 +62,9 @@ export class WalletApi extends DeveloperApi {
    * @param params the parameters for the wallet token balance request
    * @returns the token balances for the specified wallet
    */
-  async balance(params: WalletBalanceParameters): Promise<WalletTokenBalances> {
+  async balance(params: WalletBalanceParameters): Promise<WalletTokenBalance[]> {
     const { id, ...rest } = params;
-    return this.getRequest<WalletTokenBalances>(
+    return this.getRequest<WalletTokenBalance[]>(
```
Comment (krzysu) [inline packages/web3-circle-sdk/src/WalletApi.ts:67]: fyi @avkos (the return type was wrong)

File: packages/web3-circle-sdk/src/FaucetApi.ts (hunk 1 above)
Comment (avkos) [reply packages/web3-circle-sdk/src/FaucetApi.ts:11]: but it works in tests.
```
describe('Faucet Api', () => {
  const faucetApi = new FaucetApi(apikey, BASE_URL);

  it('Request native token', async () => {
    const res = await faucetApi.request({
      blockchain: BLOCKCHAIN.ETH_SEPOLIA,
      address: ETH_SEPOLIA_WALLET_ADDRESS,
      native: true,
    });
    expect(res).toBe(true);
  });

  it('Request eurc token', async () => {
    const res = await faucetApi.request({
      blockchain: BLOCKCHAIN.ETH_SEPOLIA,
      address: ETH_SEPOLIA_WALLET_ADDRESS,
      eurc: true,
    });
    expect(res).toBe(true);
  });

  it('Request usdc token', async () => {
    const res = await faucetApi.request({
      blockchain: BLOCKCHAIN.ETH_SEPOLIA,
      address: ETH_SEPOLIA_WALLET_ADDRESS,
      usdc: true,
    });
    expect(res).toBe(true);
  });
});
```

Comment (avkos) [reply packages/web3-circle-sdk/src/FaucetApi.ts:11]: our sdk return boolean

**Review by avkos — APPROVED** [review]

File: packages/web3-circle-sdk/src/WalletApi.ts (hunk 2 above)
Comment (avkos) [reply packages/web3-circle-sdk/src/WalletApi.ts:67]: fixed

File: packages/web3-circle-sdk/src/FaucetApi.ts (hunk 1 above)
Comment (krzysu) [reply packages/web3-circle-sdk/src/FaucetApi.ts:11]: I tested it from just a simple node script and faucet works fine, so it must be something with remix then, I will investigate, sorry. 
you can keep boolean return, there was a bug with 'wallet' part though in here, no?

```
------------------------------------------------------ ⬇️ (unnecessary)
await this.postRequest<void>('/faucet/drips', params, 'wallets')
```

**Pull request merged**